bluepysnap/circuit_validation.py

```diff
@@ -316,44 +312,44 @@ def validate_node_population(nodes_file, population_dict, name):
         list: List of errors, empty if no errors
     """
     with h5py.File(nodes_file, "r") as h5f:
-        nodes_h5 = h5f.get("nodes")
-
-        if not nodes_h5 or len(nodes_h5) == 0:
+        if "nodes" not in h5f or len(h5f["nodes"]) == 0:
             return []
 
-        if name not in nodes_h5:
+        # special case in which there are populations but not the one expected
+        if name not in h5f["nodes"]:
             return [fatal(f"population '{name}' not found in {nodes_file}")]
 
-        population = nodes_h5[name]
-        group = population.get("0")
+        population = h5f[f"nodes/{name}"]
 
-        if group and population.get("node_type_id"):
+        if "0" in population and "node_type_id" in population:
+            group = population["0"]
             group_df = _nodes_group_to_dataframe(group, population)
-            return _check_nodes_group(group_df, group, population_dict, name)
+            if len(group_df) > 0:
+                return _check_nodes_group(group_df, group, population_dict, name)
 
     return []
 
 
-def _get_node_ids(node_population):
+def _get_node_ids(nodes_h5, population_name):
     """Gets node ids of node population.
 
     Args:
-        node_population (h5py.Group): node population h5 instance
+        nodes_h5: (h5py.File): nodes file h5 instance
+        population_name (str): node population name
 
     Returns:
         np.ndarray: Numpy array of node ids, empty if couldn't find any
     """
-    node_ids = np.empty(0)
-    if "node_id" in node_population:
-        node_ids = node_population["node_id"][:]
-    else:
-        grp = node_population.get("0")
-        if grp:
-            for attr in grp:
-                if isinstance(grp[attr], h5py.Dataset):
-                    node_ids = np.arange(len(grp[attr]))
-                    break
-    return node_ids
+    node_population = nodes_h5.get(f"nodes/{population_name}")
```

Comment:
I still find not using  `nodes_h5["nodes"][population_name]` jarring

Reply:
I will change this